packages/utils/src/subgraphHealth.ts

```diff
@@ -38,9 +51,21 @@ type SubgraphHealth = {
  *   - "failed": subgraph halted due to errors
  * - synced: whether the subgraph is synced to the network
  */
-export const getSubgraphHealth = async (subgraphName: string): Promise<SubgraphHealth | undefined> => {
+
+const cache: { [url: string]: { record: SubgraphHealth; timestamp: number } } = {};
+export const getSubgraphHealth = async (subgraphName: string, url: string): Promise<SubgraphHealth | undefined> => {
+  const healthUrl = GET_SUBGRAPH_HEALTH_URL(url);
+  if (!healthUrl) {
+    return undefined;
+  }
+
+  // Use cache value if exist and within timeout
+  if (cache[url]?.record && Date.now() - cache[url].timestamp < 5_000) {
```

Comment:
wait wtf? we're caching twice? this is redundant with the cache check in the fallback subg

Reply:
okay will only cache at top level